espree.js

```diff
@@ -799,107 +800,120 @@ function scanNumericLiteral() {
     };
 }
 
-// 7.8.4 String Literals
+/**
+ * Scan a string escape sequence and return its special character.
+ * @param {string} ch The starting character of the given sequence.
+ * @returns {Object} An object containing the character and a flag
+ * if the escape sequence was an octal.
+ * @private
+ */
+function scanEscapeSequence(ch) {
+    var code,
+        unescaped,
+        restore,
+        escapedCh,
+        octal = false;
+
+    // An escape sequence cannot be empty
+    if (!ch) {
+        throwError({}, Messages.UnexpectedToken, "ILLEGAL");
```

Comment:
I think this case can't happen in the old code. The loop prevents reading past the end of the source text so `ch` is always valid.